### ISSUE TITLE
作者，您好。最近拜读了您的论文SemiSAM+并去学习对应的源码，在此有些小问题请教一下您，具体的内容我放在了附件里。

### ISSUE THREAD

**Issue by zed796593-cloud** [YichiZhang98/SemiSAM#14]
[document.pdf](https://github.com/user-attachments/files/26654798/document.pdf)

**Comment by YichiZhang98** [comment]
感谢您提出的问题和细致的代码审查。 1. 原compute_epistemic_uncertainty 中 torch.var(pred - ensemble)对整个 tensor 求方差返回标量，最终结果也是标量。但训练代码中需要用它做逐 voxel 加权。2. outputs_soft原代码传入了背景通道，但下游 click method 将其作为前景 mask 使用存在问题。3. SAM 的mask_decoder输出的是logits，low_res_masks 保持 logits 格式回传给 prompt_encoder 也是 SAM 的设计，但代码中logins直接传给 click_method 做判断存在问题。

我们已对代码问题进行了修复，可以参考最新的repo
